### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by bfwetmore**
# Application Full Update

Update Architecture of the entire application, with some new UI elements.

## Feature and Architectural Changes:

- [x] Move logic out of router.
- [x] Refactor logic and functions.
- [x] Breaks up that logic into a Controller and Data-Manager.
- [x] Restructure of Template files.
- [x] Add style elements via css stylesheets.
- [x] Add dark and light mode 
- [x] This change requires a documentation update
- [x] Add error handling and error template.
- [x] Results now changes Checkbox to submit form data on change.

# Unit tests and End-to-End testing

End-to-End testing involved running application on a local port and exercising each interaction a user may use with the application. Found no errors or bugs in the process.

- [x] controller_test.js - Tests functions that modifies Data for Response's - PASS
- [x] data_manager_test.js - Tests all methods that extracts and manipulates data from JSON data file.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/controllers/controller.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+/**
+ * Construct an object of users Selections.
+ * @param req
+ * @param {option? : Array} richness optional.
+ * @returns {object} of selection(s).
+ */
+function buildSelection(req, richness) {
+  const material = req.body.material ? req.body.material : req.query.material;
+  const region = req.body.region ? req.body.region : req.query.region;
+  const constellation = req.body.region
```
Comment (troglodyte) [inline src/controllers/controller.js]: This is a bit hard to read, the nested conditionals. Also, that formatting style is not really a good idea. Generally people put ternaries on one line `foo ? bar : baz`

File: src/controllers/cookieController.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * Create an array of current 'truthy' cookies.
+ * @param req
+ * @returns {array}
+ */
+function createArray(req) {
+  let cookieObject = getCookies(req);
+  let cookieArray = Object.keys(cookieObject).filter(
+    (key) => cookieObject[key]
+  );
```
Comment (troglodyte) [inline src/controllers/cookieController.js:36]: I don't think the parenthesis around `key` are necessary

File: src/controllers/cookieController.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * @returns {object}
+ */
+function checkRichness(req, richnessBody) {
+  let richnessCookie = {};
+  if (richnessBody) {
+    richnessBody.forEach((key) => (richnessCookie[key.toLowerCase()] = true));
+    return richnessCookie;
+  }
+  return getCookies(req);
+}
```
Comment (troglodyte) [inline src/controllers/cookieController.js:56]: the naming here is a little wonky. checkRichness doesn't jive with `return getCookies(req)`. 
I would suggest renaming possibly, or pulling the get cookies out and making that call outside the function

File: src/data/data-manager.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+ * @param {object} selection - Object of Selections
+ * @param {string} selection.region
+ * @param {string} selection.material
+ * @param {string} selection.constellation
+ * @param {array} selection.richnessArray
+ * @returns {array [objects]} of Systems.
+ */
+function generateResults(selection) {
+  let set = new Set();
+  piData.forEach((systemObject) => {
```
Comment (troglodyte) [inline src/data/data-manager.js:53]: while it is a 'set' my suggestion would be to name it for what it contains, not what kind of data structure it is, or you could do both, ie: SystemSet (or whatever appropriate)

File: src/router/router.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+const express = require("express");
```
Comment (troglodyte) [inline src/router/router.js:1]: Github won't open up the echoesPI.json file. Maybe it should be ignored? 🤔 (sorry i couldn't comment on that file directly)

File: src/controllers/cookieController.js (hunk 3 above)
Comment (bfwetmore) [reply src/controllers/cookieController.js:56]: Done, I found renaming the functions to jive better made more sense. Changing logic location still would have required name changing and add more logic to the controller method.

File: src/data/data-manager.js (hunk 4 above)
Comment (bfwetmore) [reply src/data/data-manager.js:53]: Great catch, I also renamed everywhere else where 'set' was used to the corresponding use of the set.

File: src/router/router.js (hunk 5 above)
Comment (bfwetmore) [reply src/router/router.js:1]: Is required for use on the server. When time allows, maybe installation scripts or even a database can be added.

File: src/controllers/cookieController.js (hunk 2 above)
Comment (bfwetmore) [reply src/controllers/cookieController.js:36]: Parenthesis removed.

File: src/controllers/controller.js (hunk 1 above)
Comment (bfwetmore) [reply src/controllers/controller.js]: Changed to be more simple and easier to read.

**Review by troglodyte — APPROVED** [review]

**Pull request merged**